src/bin/pg_autoctl/pgsetup.c

```diff
@@ -852,7 +852,18 @@ pg_setup_pgdata_exists(PostgresSetup *pgSetup)
 		return false;
 	}
 
-	return pgSetup->control.system_identifier != 0;
+	/*
+	 * Now that we know that PGDATA exists, let's grab the system identifier if
+	 * we don't have it already.
+	 */
+	if (pgSetup->control.system_identifier == 0)
+	{
+		bool missingPgdataIsOk = false;
+
+		return pg_controldata(pgSetup, missingPgdataIsOk);
```

Comment:
Is it on purpuse that we use the error boolean from `pg_controldata` as the return value from this function? The name of the function suggests that it returns true when pgdata exists. The comment of the function says that it returns true when `pg_controldata` was successful. This is a bit confusing.

Reply:
Good point, I tried to clarify the comment for the function, that was not kept up-to-date when we optimised away calls to pg_controldata, apparently.